agent/structs/config_entry_status.go

```diff
@@ -195,3 +210,467 @@ func (u *StatusUpdater) UpdateEntry() (ControlledConfigEntry, bool) {
+	// syntactically and semantically valid enough to produce some configuration
+	// in the underlying data plane. This does not indicate whether or not the
+	// configuration has been propagated to the data plane.
+	//
+	// Possible reasons for this condition to be True are:
+	//
+	// * "Accepted"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "Invalid"
+	//
+	// Possible reasons for this condition to be Unknown are:
+	//
+	// * "Pending"
+	//
+	GatewayConditionAccepted GatewayConditionType = "Accepted"
+
+	// This reason is used with the "Accepted" condition when the condition is
+	// True.
+	GatewayReasonAccepted GatewayConditionReason = "Accepted"
+
+	// This reason is used with the "Accepted" condition when the Gateway is
+	// syntactically or semantically invalid.
+	GatewayReasonInvalid GatewayConditionReason = "Invalid"
+
+	// This reason is used with the "Accepted" and "ListenersConfigured"
+	// conditions when the status is "Unknown" and no controller has reconciled
+	// the Gateway.
+	GatewayReasonPending GatewayConditionReason = "Pending"
+	// This condition indicates that the controller was able to resolve
+	// all specification requirements for all Gateway Listeners. If a Listener is
+	// conflicted, its network port should not be configured on any network
+	// elements. Even if a Listener is syntactically and semantically valid,
+	// the controller may not able to configure it on the underlying Gateway
+	// infrastructure. When setting this condition to False, a ResourceReference to
+	// the misconfigured Listener should be provided.
+
+	//
+	// Possible reasons for this condition to be True are:
+	//
+	// * "ListenersConfigured"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "ListenerHostnameConflict"
+	// * "ListenerProtocolConflict"
+	// * "ListenerPortUnavailable"
+	// * "ListenerUnsupportedProtocol"
+	// * "ListenerUnsupportedAddress"
+	//
+	// Possible reasons for this condition to be Unknown are:
+	//
+	// * "Pending"
+	//
+	GatewayConditionListenersConfigured GatewayConditionType = "ListenersConfigured"
+
+	// This reason is used with the "ListenersConfigured" condition when the
+	// condition is True.
+	GatewayReasonListenersConfigured GatewayConditionReason = "ListenersConfigured"
+
+	// This reason is used with the "ListenersConfigured" condition when
+	// the Listener conflicts with hostnames in other Listeners. For
+	// example, this reason would be used when multiple Listeners on
+	// the same port use `example.com` in the hostname field.
+	GatewayListenerReasonHostnameConflict GatewayConditionReason = "ListenerHostnameConflict"
+
+	// This reason is used with the "ListenersConfigured" condition when
+	// multiple Listeners are specified with the same Listener port
+	// number, but have conflicting protocol specifications.
+	GatewayListenerReasonProtocolConflict GatewayConditionReason = "ListenerProtocolConflict"
```

Comment:
So, for a number of these things, including this, we don't actually ever use these `GatewayConditionReason` values, namely because this is something that we can validate synchronously on an attempted write and just reject with an error. Do you mind removing all of the unused `Reason`/`Type` values that we don't actually use? Otherwise I think we're just accumulating enums that aren't actually valid for our implementation.